src/main/java/uk/gov/hmcts/reform/sscs/controller/CitizenController.java

```diff
@@ -0,0 +1,131 @@
+
+        List<OnlineHearing> casesForCitizen = citizenLoginService.findCasesForCitizen(getUserTokens(authorisation), tya);
+
+        return ResponseEntity.ok(casesForCitizen);
+    }
+
+    private IdamTokens getUserTokens(String oauth2Token) {
+        UserDetails userDetails = idamService.getUserDetails(oauth2Token);
+        return IdamTokens.builder()
+                .idamOauth2Token(oauth2Token)
+                .serviceAuthorization(idamService.generateServiceAuthorization())
+                .userId(userDetails.getId())
+                .email(userDetails.getEmail())
+                .roles(userDetails.getRoles())
+                .build();
+    }
+
+    @PostMapping(value = "{tya}")
+    @ApiOperation(value = "Associates a case with a citizen",
+            notes = "Associates a case in CCD with a citizen idam user. Checks the TYA number, email and postcode are "
+                    + "all match the case before associating the case with the idam user."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "The citizen has been associated with the case"),
+            @ApiResponse(code = 403, message = "The citizen cannot be associated with the case, either the user does not "
+                    + "exists or the email/postcode do not match the case the tya number is for."),
+    })
+    public ResponseEntity<OnlineHearing> associateUserWithCase(
+            @ApiParam(value = "user authorisation header", example = "Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdW") @RequestHeader(AUTHORIZATION) String authorisation,
+            @ApiParam(value = "tya number for an user and appeal", example = "A123-B123-c123-Dgdg") @PathVariable("tya") String tya,
+            @ApiParam(value = "email address of the appellant", example = "foo@bar.com") @RequestBody() AssociateCaseDetails associateCaseDetails
+    ) {
+        Optional<OnlineHearing> onlineHearing = citizenLoginService.associateCaseToCitizen(
+                getUserTokens(authorisation),
+                tya,
+                associateCaseDetails.getEmail(),
+                associateCaseDetails.getPostcode()
+        );
+
+        return onlineHearing
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.status(FORBIDDEN.value()).build());
+    }
+
+    @ApiOperation(value = "Log time against a case for a citizen",
+            notes = "Log time against a case a case for a citizen idam user. Checks the email is "
+                    + "match the case before logging the mya time with the idam user."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "No content, the time has been logged against the case"),
+            @ApiResponse(code = 403, message = "The time cannot be logged against the case, either the user does not "
+                    + "exists or the email do not match the case.")
```

Comment:
Minor point but the English used in this API documentation is pretty patchy - any chance you can improve please so it is clear?